calblueprint/dccentralkitchen-clerks
# Styling for sales sidebar on checkout screen + alphabet scrollbar

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

- Added an alphabetical scroll bar to the bottom of the checkout screen that allows you to scroll to the first product with a name in that letter range.
- Restyled checkout screen's sales container with (1) checkout button, (2) components with subtotal price, rewards applied, total price, and points earned, and (3) spacing between components.

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links
N/A

### Online sources
[Pretty helpful article on how to create buttons that scroll to different parts on a ScrollView](https://medium.com/@ayushi.nig/autoscroll-in-react-native-451601ac3ca8)

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs
#11 

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
Use checkout bypass to see if checkout flow works for you!
- Add several items to the cart to see if spacing on the sidebar between the cart and rewards/total price components works for you
- See if the alphabet scrollbar height matches (roughly) the checkout button height
- Should be about five columns of products on a tablet view (this is how I calculate where to scroll...)
- Test out the alphabet scroll tabs -- make sure they go to the right first items in the range

Note: Optimized for viewing on a tablet in landscape mode. Portrait mode/viewing on a phone is kinda jank rn...

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
Post-MVP plan is to make this compatible for mobile view. This would involve converting all hard-coded widths for product cards and line item cards, reworking the number of columns to display and/or playing around with text base components, and testing for visibility (since there is a lot of content on one screen as of right now).

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
Mostly end-to-end testing in terms of logging in as a clerk, looking up a customer, creating a large transaction, utilizing scrollbar tabs, and checking out. It works if the transaction shows up on Airtable!

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
![IMG_70F3E953A4CE-1](https://user-images.githubusercontent.com/21699109/77280746-625c8400-6c82-11ea-9b62-0d40cf10499b.jpeg)


[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."
